cmd/file_patch.go

```diff
@@ -167,6 +167,18 @@ If the 'values' object instead is an array, then any arrays returned by the sele
 will get the 'values' appended to them.
 `,
 		RunE: executePatch,
+		PersistentPreRunE: func(_ *cobra.Command, args []string) error {
```

Comment:
jfyi: this all could also be achieve with the following

```
patchCmd.MarkFlagsOneRequired("selector", "value")
patchCmd.MarkFlagsMutuallyExclusive("selector", "value")
```

```
$ ./deck file patch --selector foo --value foo
Error: if any flags in the group [selector value] are set none of the others can be; [selector value] were all set
```

```
$ ./deck file patch
Error: at least one of the flags in the group [selector value] is required
```

Anyway, I find the custom messages better than the default ones from cobra

Reply:
These methods are limited to flags. They won't be able to compare flags with command arguments.
In the PreRun func, we are comparing selector-value flags with patch file argument.

Reply:
Ah, I misread the issue. Ignore me :D